compiler/codegen/mli.cpp

```diff
@@ -0,0 +1,912 @@
+/*
+ * Copyright 2004-2019 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#include <cstring>
+#include "library.h"
+#include "ModuleSymbol.h"
+#include "FnSymbol.h"
+#include "beautify.h"
+#include "codegen.h"
+#include "driver.h"
+#include "expr.h"
+#include "stlUtil.h"
+#include "stringutil.h"
+#include <map>
+
+//
+// TODO:
+//
+// -- Change socket/marshal push/pull "call" generators to return statements.
+// -- Change "genDebugPrintCall" to return a statement.
+// ^^ These will allow us to clean up many newlines.
+// -- Remove explict push/pull generators and replace with calls to general.
+// ^^ This will reduce code size some more at the cost of legibility.
+// -- Remove unnecessary overloads.
+// -- Add support for the c_string type.
+//
+// CHANGELOG (4/19/2019):
+//
+// -- Remove all tabs from generated code.
+// -- Switch (beautify) to ON.
+// --
+// --
+//
+
+const char* gen_mli_marshalling = "chpl_mli_marshalling";
+const char* gen_mli_client_bundle = "chpl_mli_client_bundle";
+const char* gen_mli_server_bundle = "chpl_mli_server_bundle";
+static const char* client_main = "chpl_client.main";
+static const char* client_arg = "chpl_client.arg";
+static const char* client_res = "chpl_client.res";
+static const char* server_main = "chpl_server.main";
+static const char* server_arg = "chpl_server.arg";
+static const char* server_res = "chpl_server.res";
+static const char* marshal_push_prefix = "chpl_mli_mtpush_";
+static const char* marshal_pull_prefix = "chpl_mli_mtpull_";
+static const char* socket_push_name = "chpl_mli_push";
+static const char* socket_pull_name = "chpl_mli_pull";
+static const char* scope_begin = "{\n";
+static const char* scope_end = "}\n";
+
+class MLIContext {
+private:
```

Comment:
I don't quite follow what you mean?

Reply:
Right now it looks like:
```
class MLIContext {
private:
bool injectDebugPrintlines;
...
```
I would like it to look like:
```
class MLIContext {
   private:
      bool injectDebugPrintlines;
      ...
```

Reply:
If I jumped to line 129, it might be easy to mistake `std::string genMarshalCall(const char* s, const char* v, Type* t, bool out);` as a top-level function because it isn't indented - there's no visual indication that it is within the scope of something else, until you scroll up in the file.